middleware/auth_access_token_test.go

```diff
@@ -0,0 +1,103 @@
+package middleware
+
+import (
+	"context"
+	"testing"
+
+	grpc_auth "github.com/grpc-ecosystem/go-grpc-middleware/v2/interceptors/auth"
+	"github.com/grpc-ecosystem/go-grpc-middleware/v2/metadata"
+	grpc_test "github.com/grpc-ecosystem/go-grpc-middleware/v2/testing/testpb"
+	"github.com/stretchr/testify/suite"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	grpc_metadata "google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+)
+
+func TestAuthFuncGRPC_AccessToken(t *testing.T) {
+	ss := &TestAuthAccessTokenSuite{
+		InterceptorTestSuite: &grpc_test.InterceptorTestSuite{
+			TestService: newTestAuthenticationAccessToken(),
+			ServerOpts: []grpc.ServerOption{
+				grpc.StreamInterceptor(grpc_auth.StreamServerInterceptor(BearerAccessTokenAuthFuncGRPC(validateAccessToken))),
+				grpc.UnaryInterceptor(grpc_auth.UnaryServerInterceptor(BearerAccessTokenAuthFuncGRPC(validateAccessToken))),
+			},
+		},
+	}
+	suite.Run(t, ss)
+}
+
+func validateAccessToken(ctx context.Context, token string) error {
+	if token != "ey.LfexACqSU5qgYgp9EXSdR4rtnD7BJ0oOCNi8BKIkZ4vt25jRxyu6AXAKVNrtItb1" {
+		return status.Error(codes.Unauthenticated, "Invalid access token")
+	}
+	return nil
+}
```

Comment:
Consider making these part of the test suite and making the token a suite variable.